internal/remoteproc/remoteproc.go

```diff
@@ -17,10 +18,67 @@ const (
 	rprocFirmwareFileName     = "firmware"
 )
 
-var (
-	rprocFirmwareStorePath = rootpath.Join("lib", "firmware")
-	rprocClassPath         = rootpath.Join("sys", "class", "remoteproc")
-)
+var rprocClassPath = rootpath.Join("sys", "class", "remoteproc")
+
+func getCustomFirmwarePath() (string, error) {
+	// Check if kernel has custom firmware path configured
+	customPath, err := os.ReadFile("/sys/module/firmware_class/parameters/path")
+	if err == nil {
+		if path := strings.TrimSpace(string(customPath)); path != "" {
+			return path, nil
+		} else {
+			return "", fmt.Errorf("custom firmware path is empty")
+		}
+	} else {
+		return "", fmt.Errorf("failed to read custom firmware path /sys/module/firmware_class/parameters/path: %w", err)
+	}
```

Comment:
Checking for error then returning early is more idiomatic in Go:
https://go.dev/wiki/CodeReviewComments#indent-error-flow

```suggestion
	if err != nil {
			return "", fmt.Errorf("failed to read custom firmware path /sys/module/firmware_class/parameters/path: %w", err)
	}
	
	if path := strings.TrimSpace(string(customPath)); path != "" {
		return path, nil
	} else {
		return "", fmt.Errorf("custom firmware path is empty")
	}
```